processing/src/main/java/org/apache/druid/query/QueryContexts.java

```diff
@@ -166,6 +169,56 @@ public String toString()
     }
   }
 
+  @SuppressWarnings("unused") // Used by Jackson serialization
+  public enum RowSignatureMode
```

Comment:
What do we need this for? If it's always going to be set to `ALLOW_NULL` in typical operation (as seems the case) then it's better not to have the extra code, extra context parameter, etc.